File: frontend/src/types/ap-data-manager.ts
export type IntegrationStatus = 'healthy' | 'degraded' | 'down' | 'unknown'

export interface IntegrationHealth {
  key: string
  name: string
  category: string
  purpose: string
  status: IntegrationStatus
  measurement_method: string | null
  last_checked_at: string | null
  latency_ms: number | null
  records_seen: number
  last_activity_at: string | null
  detail: Record<string, string | number | boolean | null> | null
  last_error: string | null
}

export interface StatusCounts {
  healthy: number
  degraded: number
  down: number
  unknown: number
}

export interface DataManagerResponse {
  integrations: IntegrationHealth[]
  counts: StatusCounts
  freshness_hours: number
  partial_failure: boolean
}

File: app/schemas/ap_data_manager.py
"""Safe public response contracts for AP integration health."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel


IntegrationStatus = Literal["healthy", "degraded", "down", "unknown"]
MeasurementMethod = Literal[
    "read_only_endpoint_probe",
    "recorded_run_activity",
    "recorded_delivery_activity",
]
SafeErrorCategory = Literal[
    "authentication_failure",
    "timeout",
    "rate_limited",
    "connector_failure",
]
SafeDetailValue = str | int | float | bool | None


class IntegrationHealthSummary(BaseModel):
    key: str
    name: str
    category: str
    purpose: str
    status: IntegrationStatus
    measurement_method: MeasurementMethod | None
    last_checked_at: datetime | None
    latency_ms: int | None
    records_seen: int
    last_activity_at: datetime | None
    detail: dict[str, SafeDetailValue] | None
    last_error: SafeErrorCategory | None


class StatusCounts(BaseModel):
    healthy: int = 0
    degraded: int = 0
    down: int = 0
    unknown: int = 0


class DataManagerResponse(BaseModel):
    integrations: list[IntegrationHealthSummary]
    counts: StatusCounts
    freshness_hours: float
    partial_failure: bool
